File: threads/src/main/java/im/threads/helpers/MediaHelper.java
package im.threads.helpers;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;
import android.net.Uri;

import java.util.List;

public class MediaHelper {

    public static void grantPermissions(final Context context, final Intent intent, final Uri imageUri) {
        final List<ResolveInfo> resInfoList = context.getPackageManager().queryIntentActivities(intent, PackageManager.MATCH_DEFAULT_ONLY);
        if (!resInfoList.isEmpty()) {
            for (final ResolveInfo resolveInfo : resInfoList) {
                final String packageName = resolveInfo.activityInfo.packageName;
                context.grantUriPermission(packageName, imageUri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
                context.grantUriPermission(packageName, imageUri, Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
            }
        }
    }
}

File: threads/src/main/java/im/threads/business/utils/MediaHelper.kt
package im.threads.business.utils

import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.net.Uri

object MediaHelper {
    @JvmStatic
    fun grantPermissionsForImageUri(context: Context, intent: Intent?, imageUri: Uri?) {
        if (intent != null) {
            val resInfoList = context.packageManager.queryIntentActivities(intent, PackageManager.MATCH_DEFAULT_ONLY)
            if (resInfoList.isNotEmpty()) {
                resInfoList.forEach { resolveInfo ->
                    val packageName = resolveInfo.activityInfo.packageName
                    context.grantUriPermission(packageName, imageUri, Intent.FLAG_GRANT_READ_URI_PERMISSION)
                    context.grantUriPermission(packageName, imageUri, Intent.FLAG_GRANT_WRITE_URI_PERMISSION)
                }
            }
        }
    }
}
